gammapy/spectrum/models.py

```diff
@@ -1063,31 +1065,192 @@ def plot(self, energy_range, ax=None, energy_unit='TeV',
-            spectral model
-        table_model : `~gammapy.spectrum.models.TableModel`
-            table model
+        filename : `str`
+            File containing the 1-dimensional XSPEC model.
         """
-        self.spectral_model = spectral_model
-        self.table_model = table_model
-        # Will be implemented later for sherpa fit
-        self.parameters = ParameterList([])
 
-    def evaluate(self, energy):
-        flux = self.spectral_model.__call__(energy)
-        absorption = self.table_model.__call__(energy)
-        return flux * absorption
+        # Create EBL data array
+        filename = str(make_path(filename))
+        table_param = Table.read(filename, hdu='PARAMETERS')
 
-    def to_sherpa(self, name='default'):
-        """Convert to Sherpa model
+        par_min = table_param['MINIMUM']
+        par_max = table_param['MAXIMUM']
 
-        To be implemented by subclasses
+        par_array = table_param[0]['VALUE']
+        par_delta = np.diff(par_array)*0.5
+
+        param_lo, param_hi = par_array, par_array  # initialisation
+        param_lo[0] = par_min - par_delta[0]
+        param_lo[1:] -= par_delta
+        param_hi[:-1] += par_delta
+        param_hi[-1] = par_max
+        
+        # Get energy values
+        table_energy = Table.read(filename, hdu='ENERGIES')
+        energy_lo = table_energy['ENERG_LO'] * u.keV  # unit not stored in file
+        energy_hi = table_energy['ENERG_HI'] * u.keV  # unit not stored in file
+        
+        # Energies are in keV
+        energy_bounds = EnergyBounds.from_lower_and_upper_bounds(lower=energy_lo,
+                                                                 upper=energy_hi,
+                                                                 unit=u.keV)
+
+        # Get spectrum values
+        table_spectra = Table.read(filename, hdu='SPECTRA')
+        data = table_spectra['INTPSPEC'].data
+        
+        return cls(energy_lo=energy_bounds.lower_bounds,
+                   energy_hi=energy_bounds.upper_bounds,
+                   param_lo=param_lo, param_hi=param_hi, data=data)
+
+    @classmethod
+    def read_builtin(cls, name):
         """
-        raise NotImplementedError('{}'.format(self.__class__.__name__))
+        Read one of the built-in absorption models.
+
+        Parameters
+        ----------
+        name : {'franceschini', 'dominguez', 'finke'}
+            name of one of the available model in gammapy-extra
+        """
+        models = dict()
+        models['franceschini'] = '$GAMMAPY_EXTRA/datasets/ebl/ebl_franceschini.fits.gz'
+        models['dominguez'] = '$GAMMAPY_EXTRA/datasets/ebl/ebl_dominguez11.fits.gz'
+        models['finke'] = '$GAMMAPY_EXTRA/datasets/ebl/frd_abs.fits.gz'
+        
+        return cls.read(models[name])
+    
+    def table_model(self, parameter, unit='TeV'):
+        """
+        Returns `~gammapy.spectrum.models.TableModel` for a given parameter
+        from the input aborbed model
```

Comment:
aborbed -> absorbed.

Generally, try to fit the description on the first line. End it with a dot. Then if needed a blank line and more text after that. Fill words like "Returns" can always be omitted. The return type can be listed in a `Returns` Section after `Parameters`.